march_rqt_gait_generator/src/march_rqt_gait_generator/model/modifiable_joint_trajectory.py

```diff
@@ -73,31 +73,21 @@ def duration(self, duration):
     def get_interpolated_position(self, time):
         for i in range(0, len(self.interpolated_setpoints[0])):
             if self.interpolated_setpoints[0][i] > time:
-                return self.interpolated_setpoints[1][i - 1]
+                return self.interpolated_setpoints[1][i]
 
         return self.interpolated_setpoints[1][-1]
 
     def enforce_limits(self):
         self.setpoints[0].time = 0
         self.setpoints[-1].time = self.duration
 
-        for i in range(0, len(self.setpoints)):
-            self.setpoints[i].position = min(max(self.setpoints[i].position,
-                                                 self.limits.lower),
-                                             self.limits.upper)
-            self.setpoints[i].velocity = min(max(self.setpoints[i].velocity,
-                                                 -self.limits.velocity),
-                                             self.limits.velocity)
-
-    def within_safety_limits(self):
-        for i in range(0, len(self.interpolated_setpoints)):
-            if self.interpolated_setpoints[i].position > self.limits.upper or \
-                    self.interpolated_setpoints[i].position < self.limits.lower:
-                return False
-            if i > 0 and abs(
-                    self.interpolated_setpoints[i] - self.interpolated_setpoints[i - 1]) > self.limits.velocity:
-                return False
-            return True
+        for setpoint in self.setpoints:
+            setpoint.position = min(max(setpoint.position,
+                                        self.limits.lower),
+                                    self.limits.upper)
+            setpoint.velocity = min(max(setpoint.velocity,
+                                        -self.limits.velocity),
+                                    self.limits.velocity)
```

Comment:
Write both  in one line, easier to read in my opinion.

Reply:
I prefered it this way because it is easier to see which belong to the max and which to the min. Maybe we can toss a coin ;)?

Reply:
opinions differ, doesn't effect the functionalities so lets keep it like this.